### PR TITLE
Fix crash when an attribute node was used inside an unary op

### LINKED ISSUE #8207: Crash:  AstroidError in refactoring_checker.py
### Bug description

proprietary code, I haven't tried to narrow it down

pylint crashed with ``AstroidError`` and with the following stacktrace:
```
Traceback (most recent call last):
  File "/.../python3.8/site-packages/pylint/lint/pylinter.py", line 798, in _lint_file
    check_astroid_module(module)
  File "/.../python3.8/site-packages/pylint/lint/pylinter.py", line 1067, in check_astroid_module
    retval = self._check_astroid_module(
  File "/.../python3.8/site-packages/pylint/lint/pylinter.py", line 1117, in _check_astroid_module
    walker.walk(node)
  File "/.../python3.8/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  File "/.../python3.8/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  File "/.../python3.8/site-packages/pylint/utils/ast_walker.py", line 94, in walk
    self.walk(child)
  [Previous line repeated 2 more times]
  File "/.../python3.8/site-packages/pylint/utils/ast_walker.py", line 91, in walk
    callback(astroid)
  File "/.../python3.8/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 1731, in visit_comprehension
    self._check_unnecessary_list_index_lookup(node)
  File "/.../python3.8/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 2208, in _check_unnecessary_list_index_lookup
    has_start_arg, confidence = self._enumerate_with_start(node)
  File "/.../python3.8/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 2326, in _enumerate_with_start
    start_val, confidence = self._get_start_value(start_arg)
  File "/.../python3.8/site-packages/pylint/checkers/refactoring/refactoring_checker.py", line 2346, in _get_start_value
    return node.operand.value, HIGH
AttributeError: 'Attribute' object has no attribute 'value'

The above exception was the direct cause of the following exception:

Traceback (most recent call last):
  File "/.../python3.8/site-packages/pylint/lint/pylinter.py", line 763, in _lint_files
    self._lint_file(fileitem, module, check_astroid_module)
  File "/.../python3.8/site-packages/pylint/lint/pylinter.py", line 800, in _lint_file
    raise astroid.AstroidError from e
astroid.exceptions.AstroidError
```

### Configuration

_No response_

### Command used

```shell
pylint
```

### Pylint output

```shell
n/a
```

### Expected behavior

n/a

### Pylint version

```shell
pylint 2.16.1
astroid 2.14.1
Python 3.8.12 (default, Sep 23 2021, 08:42:37)
[Clang 12.0.0 (clang-1200.0.32.29)]
```

### OS / Environment

_No response_

### Additional dependencies

_No response_

### PR DESCRIPTION
**Pull request by Pierre-Sassoulas**
## Type of Changes

|     | Type                   |
| --- | ---------------------- |
| ✓   | :bug: Bug fix          |

## Description

Closes #8207

### CODE REVIEW

File: pylint/checkers/refactoring/refactoring_checker.py [hunk 1]
```diff
@@ -2338,13 +2338,17 @@ def _enumerate_with_start(
...
+        ):
             inferred = utils.safe_infer(node)
             start_val = inferred.value if inferred else None
             return start_val, INFERENCE
         if isinstance(node, nodes.UnaryOp):
             return node.operand.value, HIGH
         if isinstance(node, nodes.Const):
             return node.value, HIGH
-
+        print(node.operand.__dict__, node.operand.as_string(), type(node.operand))
```
Comment (Pierre-Sassoulas) [inline pylint/checkers/refactoring/refactoring_checker.py:2353]:
```suggestion
```

**Review by clavedeluna — CHANGES_REQUESTED** [review]
I was just writing a PR to fix this since I feel responsible for writing it! The real fix turns out to be
```
    def _get_start_value(self, node: nodes.NodeNG) -> tuple[int | None, Confidence]:
        inferred = utils.safe_infer(node)
        start_val = inferred.value if inferred else None
        return start_val, INFERENCE
```

when I initially wrote it I was just not as familiar with when to use inference so I was super conservative, but turns out this should handle all cases!

**Comment by clavedeluna** [comment]
You'll also need to update the output .txt with my suggestion

**Comment by Pierre-Sassoulas** [comment]
Isn't it better to not use inference all the time for performance ?

**Comment by clavedeluna** [comment]
> Isn't it better to not use inference all the time for performance ?

You'd know better than I :) but I guess here we're trading performance for preventing another one of these bugs in the future in case there is another type of `node` we didn't think of in the tests.

**Comment by Pierre-Sassoulas** [comment]
I think at this point we pretty much covered all the crash cases, the issues are getting increasingly niche. But yeah it crashed a lot since initial implementation. 😄 What do you think ? 

> You'll also need to update the output .txt with my suggestion

I did not understand that part ?

**Comment by clavedeluna** [comment]
> I think at this point we pretty much covered all the crash cases, the issues are getting increasingly niche. But yeah it crashed a lot since initial implementation. 😄 What do you think ?
> 
> > You'll also need to update the output .txt with my suggestion
> 
> I did not understand that part ?

No need to update anything since we've discussed it.

**Review by clavedeluna — APPROVED** [review]

**Comment by Pierre-Sassoulas** [comment]
I have no idea if the performance are better with the current code compared to inference tbh. And what you proposed looks a lot better and also fail proof.

**Comment by Pierre-Sassoulas** [comment]
@DanielNoord do you have an opinion on the subject ?

**Pull request merged**